src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -67,10 +74,20 @@ function initStore(appStore: AppStore) {
                 getBrowserWindow().globalStores.routing.location.pathname,
             ];
         },
-        (x: any) => {
-            const query = x[0] as CancerStudyQueryUrlParams;
+        (x:any) => {
+
+
+
+            // Genes exist means initStore is called to initiate a secondary store that will be used by pathway mapper
```

Comment:
I'm not convinced that making a duplicate copy of the store is the right approach.   I understand why you did this (need alteration data for non-queried genes).    Can you list exactly what parts of the resultsview store are consumed by the PathwayMapper library?  Maybe the mobxpromise (remotedata) could be pulled out and so shared between instances.   Would be good to discuss.

Reply:
I thought it would be a inefficient solution but still I thought it was the only solution. If we can find an alternative way to achieve this, that would be great. PathwayMapper expects only alteration frequency data. It is calculated by that [function](https://github.com/Rgtemze/cbioportal-frontend/blob/eeae6bec2f517e04f7b987c29a4ee4d7298ffc40/src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx#L109) which is implemented [here](https://github.com/Rgtemze/cbioportal-frontend/blob/eeae6bec2f517e04f7b987c29a4ee4d7298ffc40/src/shared/components/oncoprint/OncoprintUtils.ts#L445).